ext/random/csprng.c

```diff
@@ -147,19 +154,17 @@ PHPAPI zend_result php_random_bytes(void *bytes, size_t size, bool should_throw)
 	}
 # endif
 	if (read_bytes < size) {
-		int    fd = RANDOM_G(random_fd);
+		int    fd = zend_atomic_int_load_ex(&random_fd);
```

Comment:
Replaced the thread-specific global by a true global, so that we can use `php_random_bytes_ex()` earlier.